src/objects/obstacles/cloud.ts

```diff
@@ -1,28 +1,94 @@
-
-import { BaseObject } from '../../utils/interfaces/object-abstract-class.ts';
-import * as CONFIG from '../../utils/configuration.ts'
+import { BaseObject } from "../../utils/interfaces/object-abstract-class.ts";
+import * as CONFIG from "../../utils/configuration.ts";
+import { generateRandom } from "../../utils/generateRandom.ts";
+import { Game } from "../../scenes/Game.ts";
 
 export class Cloud extends BaseObject {
+  private cloudGroup: Phaser.Physics.Arcade.Group;
+  private pipeListener: Function;
 
-    constructor(scene: Phaser.Scene) {
-    
-        super(scene);
-        this.init();  
-    
-    }
+  constructor(scene: Phaser.Scene) {
+    super(scene);
+    this.init();
+    this.pipeListener = this.GenerateNewCloud.bind(this);
```

Comment:
so while i prefer this syntax, I think everywhere else we've been using `GenerateNewCloud = () => {}`
arrow functions essentially do the same as bind(this)

I think consistency is good, but readability is better, so i'll leave it up to you to the preference.